src/sql/workbench/contrib/editData/browser/editDataGridPanel.ts

```diff
@@ -1189,7 +1214,7 @@ export class EditDataGridPanel extends GridParentComponent {
 		let valueToDisplay = '';
 		let cellClasses = 'grid-cell-value-container';
 		/* tslint:disable:no-null-keyword */
-		let valueMissing = value === undefined || value === null || (Services.DBCellValue.isDBCellValue(value) && value.isNull) || value === 'NULL';
+		let valueMissing = value === undefined || value === null || (Services.DBCellValue.isDBCellValue(value) && (value.isNull || value.displayValue.length === 0)) || value === 'NULL';
```

Comment:
An empty value isn't necessarily a null value though - what about an empty string?

Reply:
Well currently, whatever is returned by the service does not distinguish between null or empty when the row is added, so in order to compensate for this, we will need to check if a new row is being added or not during the refresh.

Reply:
Just added logic in the load data function to determine if a newly added row contains null values or not.